src/Couchbase.Lite.Shared/API/Sync/ReplicatorConfiguration.cs

```diff
@@ -152,7 +152,14 @@ public IList<string> Channels
         public bool Continuous
         {
             get => _continuous;
-            set => _freezer.SetValue(ref _continuous, value);
+            set {
+                try {
+                    _freezer.SetValue(ref _continuous, value);
+                    Options.MaxRetries = _continuous ? Int32.MaxValue : 9;
+                } catch {
```

Comment:
What is this catching?

Reply:
This is to take care the case when we alter the Continuous value in replicator configuration after replicator is created internally. 
```var config = CreateConfig(true, false, false);
            using (var repl = new Replicator(config)) {
                repl.Config.MaxRetries.Should().Be(9, "Because default Max Retries is 9 times for a Single Shot Replicator.");
                repl.Config.Continuous = true; // Internally
                repl.Config.MaxRetries.Should().Be(9, "Because default Max Retries is 9 times for a Single Shot Replicator.");
            }```

Reply:
I don't understand.  What is the goal here?  Now the user will not know if they did something wrong.  Furthermore, if they set max retries before this will override it.

Reply:
Now the user will not be alerted if they do something wrong.  What's wrong with not trying and catching?

Reply:
Logic removed.